opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -84,7 +84,7 @@
 logger = getLogger(__name__)
 
 # TODO: quarantine
-ParentSpan = typing.Optional[typing.Union["Span", "SpanContext"]]
+Parent = typing.Optional[typing.Union["SpanContext", "Span"]]
```

Comment:
Is this union still needed here if the Parent is always a SpanContext?

Reply:
According to https://github.com/open-telemetry/opentelemetry-specification/blob/c04041e16c9571b7ee5cfd5f102fc15e94ffacd7/specification/trace/api.md#span-creation, the API must also accept a Span as parent. In any case, it would be a needless breaking and convenience-decreasing change to forbid this.

Reply:
Yes, I thought through whether to bring this up in the spec as well, but I think there's a good rationalle for letting the API support this: it's a lot more intuitive for those who want to start new spans themselves, since most people only interact with Spans.

Reply:
no value in the swap... I do think there's value in the name.

But thinking through it, this is a backwards incompatible API change, and we really shouldn't be doing those needlessly. I'll revert that.